### DISCUSSION TITLE
Add "send now" option

### DISCUSSION THREAD

**Discussion by martinmagnusson** [Extended-Thunder/send-later#316] [Ideas]
Sometimes I have a batch of emails queued for being sent later at some point, but I would like to send them right away (maybe because I know I will be offline at the scheduled time so that they cannot be sent then, or some other reason). It would be great if there were a "send queued messages now" option (for example, in the popup menu at "Send Later [PEND N]").

**Comment by jph-sendlater** [comment]
Normally you would handle this situation by opening queued messages from the drafts folder and sending each one using the regular "send" button. Are you dealing with lots of messages, such that handling each one separately would be a problem?

I like this idea (especially because I think the status bar menu is under-utilized), but it has some practical problems that I think might make it a non-starter. For instance, how would recurring messages be handled?

**Reply by martinmagnusson** [reply]
Right, that would be a possible workaround. I might have 10 or so messages, so it's not a very hard problem, but a bit of a nuisance. I didn't think about recurring messages (I don't use that feature). Maybe this feature could be limited to only non-recurring messages?

**Reply by jph-sendlater** [reply]
That wouldn't be very intuitive from a UI perspective. I'll think more about this, though. I intend to eventually introduce an alternative message storage mechanism that does not use the draft folders, and maybe I can come up with a way to work something like this into that new system.

**Reply by rgrun1** [reply]
I delay messages by a minute to avoid having to wait on each email to connect and be delivered
I would love to see  something (a button?) that would override that and send immediately

**Reply by jph-sendlater** [reply]
There's always the 'send now' button in the scheduler dialog. Is that not what you're looking for?

**Reply by Spokey42** [reply]
Not sure I understand.  What is the scheduler dialog?  The thing in the status bar?  I don't see any send now there.  I too would want this capability but only need it occasionally so if there is something easy right now, that is OK for me.

**Comment by rgrun1** [comment]
Well {blush} ... that should do it

On 4/28/2021 3:30 PM, jph-sendlater wrote:
>
> There's always the 'send now' button in the scheduler dialog. Is that 
> not what you're looking for?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub 
> <https://github.com/Extended-Thunder/send-later/discussions/316#discussioncomment-671728>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AA5ED5WCUNSE3LFDICLY7ULTLBO6BANCNFSM43JURJOA>.
>
--

**Reply by jph-sendlater** [reply]
No worries; I'm happy to help.

**Comment by rgrun1** [comment]
BTW Thanks!!

On 4/28/2021 3:30 PM, jph-sendlater wrote:
>
> There's always the 'send now' button in the scheduler dialog. Is that 
> not what you're looking for?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub 
> <https://github.com/Extended-Thunder/send-later/discussions/316#discussioncomment-671728>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AA5ED5WCUNSE3LFDICLY7ULTLBO6BANCNFSM43JURJOA>.
>
--

**Comment by FaBay89** [comment]
A "Send all now" button would be really nice, you could use it when you want to close thunderbird and there are still mails unsent

**Comment by jikamens** [comment]
This enhancement is being tracked in #485.

**Comment by jikamens** [comment]
I think the newly added feature to allow messages in the Drafts folder to be selected and scheduled addresses this user story. You can select all the messages you want to send now, right-click on them, select Send Later > Schedule Message(s) from the menu, and click the Send Now button to send them all.
